### PR TITLE
Use new loki API

### PR DESCRIPTION
**Pull request by ctrox**
Switch to the new loki API As the old prom API is [deprecated](https://github.com/grafana/loki/blob/master/docs/api.md#get-apipromquery).

This uses the client from the logcli package so we don't need to worry about marshaling/unmarshaling anymore. Regexp filtering via parameter is not possible anymore using the new API so that field has been removed. Filtering should be done directly in the query with logql.

I have tested this locally and it produced the same metrics as the previous version, except that the `http_*` metrics are gone. So far I have not been able to get those back with the custom registry. Let me know if you have an idea about that or if we should just leave it as is.

### CODE REVIEW

File: cmd/loki_exporter/loki_exporter.go [hunk 1]
```diff
@@ -48,11 +49,16 @@ func main() {
...
 
-	prometheus.MustRegister(exporter)
-	prometheus.MustRegister(version.NewCollector("loki_exporter"))
+	// use our own registry to not get metrics from loki dependencies like etcd
+	r := prometheus.NewRegistry()
+	r.MustRegister(exporter)
+	r.MustRegister(version.NewCollector("loki_exporter"))
+	r.MustRegister(prometheus.NewGoCollector())
+	r.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	handler := promhttp.HandlerFor(r, promhttp.HandlerOpts{})
```
Comment (ricoberger) [inline cmd/loki_exporter/loki_exporter.go]: Maybe you can add the [InstrumentMetricHandler](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp?tab=doc#InstrumentMetricHandler):

```suggestion
	handler := promhttp.InstrumentMetricHandler(r, promhttp.HandlerFor(r, promhttp.HandlerOpts{}))
```

**Comment by ricoberger** [comment]
Hi, thanks for your PR, what a incredible work.

I think the `http_*` metrics are not really needed. It would be enough to add the the `InstrumentMetricHandler`, which produces the following metrics for the `/metrics` ednpoint:

```
# HELP promhttp_metric_handler_requests_in_flight Current number of scrapes being served.
# TYPE promhttp_metric_handler_requests_in_flight gauge
promhttp_metric_handler_requests_in_flight 1
# HELP promhttp_metric_handler_requests_total Total number of scrapes by HTTP status code.
# TYPE promhttp_metric_handler_requests_total counter
promhttp_metric_handler_requests_total{code="200"} 0
promhttp_metric_handler_requests_total{code="500"} 0
promhttp_metric_handler_requests_total{code="503"} 0
```

File: cmd/loki_exporter/loki_exporter.go (hunk 1 above)
Comment (ctrox) [reply cmd/loki_exporter/loki_exporter.go]: Thanks for that, I added it.

**Comment by ricoberger** [comment]
Thank you. I merge this and create a new version.

**Pull request merged**